### DISCUSSION TITLE
MRTK3 Preview 15 Release

### DISCUSSION THREAD

**Discussion by AMollis** [microsoft/MixedRealityToolkit-Unity#11536] [Releases]
![MRTK3_Release_Image_v3_0_0_pre_15](https://user-images.githubusercontent.com/36461279/236016972-9ed56869-d669-4968-a341-6142e3adbf56.png)

We’re thrilled to announce the pre.15 release for MRTK3! Our new team has been working hard to ramp up and continue the amazing work started by the original MRTK3 team.

In this release, we’ve made good progress towards MRTK3 GA. We’ve updated MRTK3’s minimum Unity version to 2021.3 LTS, as well as updated our XRI dependency to version 2.3 and AR Foundation dependency to version 5.0. We’ve also added See-It Say-It labels to button prefabs and made some additional sample scene improvements.

We’re excited to hear from the community about these changes and look forward to continuing the journey to MRTK3 GA.

# Changelog for MRTK3 pre.15

## [pre.15] - 2022-4-28 - [`41a7766`](https://github.com/microsoft/MixedRealityToolkit-Unity/commit/41a77667c6d514a40b1269c6256b248b97539545)

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/). The content is mirrored at [the changelog page of the Microsoft docs website](https://learn.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/mrtk3-overview/changelogs/mrtk3-pre-15-changelog), for now.

### Added

- Added RequiresHover field to StatefulInteractable ([#11394](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11394))
- Added TransformExtensions class and unit tests ([#11435](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11435))
- Added See-It Say-It Example Scene ([#11446](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11446))
- Add support for custom voices on WindowsTextToSpeechSubsystem ([#11458](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11460))
- Added See-It Say-It Labels To Buttons ([#11466](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11466))
- Added Min Max Scale Constraints to Sample Scenes and Fixed Inspector Highlight ([#11497](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11497))
- Added Additional Sample Scenes ([#11522](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11522))
  
### Changed

- Removed DrawIf usage from AnimationEffect ([#11438](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11438))
- Updated MRTK3's minimum supported XRI version XRI 2.3 ([#11452](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11452))
- Changed SpeechInteractor so voice commands enabled when an interactable isHovered ([#11462](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11462))
- Removing VisualProfiler from MRTK3, in favor of the one available at https://github.com/microsoft/VisualProfiler-Unity.git ([#11473](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11473))
- Upgrade MRTK3's minimum Unity version to 2021 LTS ([#11465](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11465))

### Fixed

- Switch hands in SolverHandler when interactor is not actively hovering (
[#11441](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11441))
- Lock position of hand when rotating in simulator ([#11457](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11457)) 
- Fix input simulator Face Camera action ([#11460](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11460)) 
- Updated sample scene to support spatial audio ([#11469](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11469))
- Fixed Constraints Inspector Highlight ([#11497](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11497))
- Fixed two hand object manipulation of type scaling not scaling the right way in MRTK3 ([#11516](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11516))
- Fixed Tap-to-Place Slover ([#11522](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11522))

**Comment by amccraw** [comment]
Is there an updated target date for general availability for MRTK3?

**Reply by AMollis** [reply]
@amccraw , unfortunately we can't share a timeframe yet.  However, soon we will provide the community with our updated GA plans.

**Comment by LocalJoost** [comment]
Wow, how could I have missed this! I do have a question though - even though I refreshed the feed, the new packages are not showing up in the MRTK feature tool

**Reply by SimonDarksideJ** [reply]
Would still be much better to follow the same standard practice other toolkits use and publish the packages (where possible) to OpenUPM.  That would make access to the toolkit MUCH better / easier.

**Reply by AMollis** [reply]
@LocalJoost , thank you for your comment.  Are you still not seeing "pre.15" in the feature tool? 
<img width="376" alt="image" src="https://github.com/microsoft/MixedRealityToolkit-Unity/assets/36461279/de8f15f9-3519-4c0f-92cf-76d5fdedbb1a">

@SimonDarksideJ , thank you as well.  We are aware of the difficulties with the MR Feature Tool.  We are working on ways to improve the developer story.  Nothing to announce yet, but this is something we're aware of and working on.

**Reply by LocalJoost** [reply]
@AMollis Yes it does. I found the problem - I checked while using a 2020 project. And it says, clearly, in the last line of the change list:
"Upgrade MRTK3's minimum Unity version to 2021 LTS". 
In a 2021.3 project it shows up fine. Ah well :D

**Reply by SimonDarksideJ** [reply]
Then ideally the Feature Tool should display a warning or information box to that effect, if Feature 14 is the maximum for a version of Unity.
Avoids confusion.

**Comment by Asanche** [comment]
I noticed as part of this release (which was a while ago), the dependent version of ARFoundation for the com.microsoft.mrtk.input module increased from 4.2.6 to 5.0.5. These two versions have drastically different namespace structures (specifically how they distribute the arsubsystem namespace) and could result in needing to upgrade a lot of other dependencies. I don't see a mention of this in the notes, was this an intentional change?

**Reply by AMollis** [reply]
Yes this was intentional.  This specific change was called out in the PR linked in this bullet point:

* Upgrade MRTK3's minimum Unity version to 2021 LTS (https://github.com/microsoft/MixedRealityToolkit-Unity/pull/11465)

We unfortunately neglected to call out the ARFoundation upgrade in the release notes.
